src/CsvTextEditor/ViewModels/MainViewModel.cs

```diff
@@ -35,34 +39,23 @@ public MainViewModel(IProjectManager projectManager)
         public Project Project { get; set; }
 
         #region Methods
-        protected override Task InitializeAsync()
+        protected override async Task InitializeAsync()
         {
-            _projectManager.ProjectActivatedAsync += OnProjectActivated;
+            await base.InitializeAsync();
 
-            return base.InitializeAsync();
+            _projectManager.ProjectActivationAsync += OnProjectActivationAsync;
         }
 
-        protected override Task CloseAsync()
+        private async Task OnProjectActivationAsync(object sender, ProjectUpdatingCancelEventArgs e)
         {
-            _projectManager.ProjectActivatedAsync -= OnProjectActivated;
-
-            return base.CloseAsync();
+            _dispatcherService.Invoke(() => Project = e.NewProject as Project, true);
```

Comment:
cause i don't know what actually comes from eventArgs. hm... may be i shouldn't even bother VM if project have different type. Something like:
  private async Task OnProjectActivationAsync(object sender, ProjectUpdatingCancelEventArgs e)
  {
       var project = e.NewProject as Project;
       if(project == null)
       {
           return;
       }       
        _dispatcherService.Invoke(() => Project = project, true);
but this time i eliminate a term when project is null... Actually a don't understand why it's so important not to use a save cast here?... As i understand this is not so frequent operation and between casting and save casting there is no performance difference...  may be i don't get it?  I think if we introduce a Project property with type 'Project' not 'IProject' - we have a contract that we should support it